core/vibes/soul/primitives/navigation/index.tsx

```diff
@@ -824,26 +822,26 @@ function SearchResults({
   );
 }
 
-function LocaleForm({
-  action,
+function LocaleSwitcher({
+  onChange,
   locales,
   activeLocaleId,
 }: {
   activeLocaleId?: string;
-  action: LocaleAction;
+  onChange: (localeId: string) => void;
   locales: [Locale, ...Locale[]];
 }) {
-  const [lastResult, formAction] = useActionState(action, null);
   const activeLocale = locales.find((locale) => locale.id === activeLocaleId);
-
-  useEffect(() => {
-    if (lastResult?.error) console.log(lastResult.error);
-  }, [lastResult?.error]);
+  const [isPending, startTransition] = useTransition();
```

Comment:
`useTransition` allows us to show an affordance while the transition is in progress.